include/linker/common-rom.ld

```diff
@@ -155,6 +155,15 @@
 	} GROUP_LINK_IN(ROMABLE_REGION)
 #endif /* CONFIG_EMUL */
 
+#if defined(CONFIG_ZTEST)
+	SECTION_DATA_PROLOGUE(ztest_section,,)
+	{
+		__ztest_suite_start = .;
+		KEEP(*(SORT_BY_NAME(".ztest_suites")));
+		__ztest_suite_end = .;
+	} GROUP_LINK_IN(ROMABLE_REGION)
+#endif /* CONFIG_ZTEST */
```

Comment:
Since `metadata`, which resides in this section, is mutable, we must make this a RAM section.

Reply:
Thanks for pointing me in the right direction, can't say I'm a wiz on linker scripts :)